src/Silex/ControllerCollection.php

```diff
@@ -57,11 +59,21 @@ public function __construct(Route $defaultRoute, $routesFactory = null)
     /**
      * Mounts controllers under the given route prefix.
      *
-     * @param string               $prefix      The route prefix
-     * @param ControllerCollection $controllers A ControllerCollection instance
+     * @param string                        $prefix      The route prefix
+     * @param ControllerCollection|callable $controllers A ControllerCollection instance or a callable for defining routes
+     *
+     * @throws \LogicException
      */
-    public function mount($prefix, ControllerCollection $controllers)
+    public function mount($prefix, $controllers)
     {
+        if (is_callable($controllers)) {
+            $collection = $this->controllersFactory ? call_user_func($this->controllersFactory) : new static(new Route(), new RouteCollection());
+            call_user_func($controllers, $collection);
+            $controllers = $collection;
```

Comment:
@fabpot, I just looked at how the [Slim Framework](http://www.slimframework.com/docs/objects/router.html#route-groups) does it. There implementation is that great either, so I think we could get away with just doing the update in the application mount and leave the ControllerCollection class alone. But document it clearly..